pkg/services/govmomi/vcenter/clone.go

```diff
@@ -390,6 +408,117 @@ func getDiskConfigSpec(disk *types.VirtualDisk, diskCloneCapacityKB int64) (type
 	}, nil
 }
 
+// createDataDisks parses through the list of VSphereDisk objects and generates the VirtualDeviceConfigSpec for each one.
+func createDataDisks(ctx context.Context, disks []infrav1.VSphereDisk, primaryDisk *types.VirtualDisk, devices object.VirtualDeviceList) ([]types.BaseVirtualDeviceConfigSpec, error) {
+	log := ctrl.LoggerFrom(ctx)
+	additionalDisks := []types.BaseVirtualDeviceConfigSpec{}
+	unit := int32(1)
+
+	for i, dataDisk := range disks {
+		log.Info("Adding disk", "name", dataDisk.Name, "spec", dataDisk)
+
+		// Get the controller of the primary disk.
+		controller, ok := devices.FindByKey(primaryDisk.ControllerKey).(types.BaseVirtualController)
+		if !ok {
+			return nil, errors.Errorf("unable to find controller with key=%v", primaryDisk.ControllerKey)
+		}
+
+		dev := &types.VirtualDisk{
+			VirtualDevice: types.VirtualDevice{
+				Key: devices.NewKey() - int32(i),
+				Backing: &types.VirtualDiskFlatVer2BackingInfo{
+					DiskMode:        string(types.VirtualDiskModePersistent),
+					ThinProvisioned: types.NewBool(true),
+					VirtualDeviceFileBackingInfo: types.VirtualDeviceFileBackingInfo{
+						FileName: "",
+					},
+				},
+				ControllerKey: controller.GetVirtualController().Key,
+			},
+			CapacityInKB: int64(dataDisk.SizeGiB) * 1024 * 1024,
+		}
+
+		// Assign unit number to the next slot on the controller.
+		if err := assignUnitNumber(ctx, dev, devices, additionalDisks, controller, unit); err != nil {
+			log.Error(err, "unable to assign unit number")
+			return nil, err
```

Comment:
```suggestion
			return nil, errors.Wrap(err, "failed to assign device unit number to disk")
```

Errors are logged by controller runtime, having the log here would log it twice.

Reply:
Ok, I will update this. Thanks!